src/Admin/WebHookAdmin.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace AppBundle\Admin;
+
+use AppBundle\Entity\OAuth\Client;
+use AppBundle\WebHook\WebHookManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\Extension\Core\Type\UrlType;
+
+class WebHookAdmin extends AbstractAdmin
+{
+    private $manager;
+    private $em;
+
+    public function __construct(
+        string $code,
+        string $class,
+        string $baseControllerName,
+        WebHookManager $manager,
+        EntityManagerInterface $em
+    ) {
+        parent::__construct($code, $class, $baseControllerName);
+
+        $this->manager = $manager;
+        $this->em = $em;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDatagrid()
+    {
+        if (!$this->datagrid) {
+            $this->datagrid = new WebHookDataGrid(parent::getDatagrid(), $this->manager, $this->em);
+        }
+
+        return $this->datagrid;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureRoutes(RouteCollection $collection)
+    {
+        $collection->clearExcept(['list', 'edit']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureFormFields(FormMapper $form)
+    {
+        $form
+            ->add('event', TextType::class, [
+                'disabled' => true,
+                'label' => 'Event Name',
+            ])
+            ->add('client', EntityType::class, [
+                'disabled' => true,
+                'class' => Client::class,
+                'label' => 'Client',
+            ])
+            ->add('callbacks', CollectionType::class, [
+                'label' => 'Liste des callbacks',
+                'entry_type' => UrlType::class,
+                'allow_add' => true,
+                'allow_delete' => true,
```

Comment:
You should be careful with this when using json in the model, see https://github.com/symfony/symfony/issues/23653.